stackrox/stackrox
# ROX-18978, ROX-20225: Policy updates

## Description

Combining two changes into one PR because they all deal with one migration.

The changes include:

* Update the policy migrator code to make it a little more efficient (avoid an unnecessary Exists call). Unfortunately this does lead to some code duplication, but given that the code is used in old migrations, it's best to be careful.
* Some policies have duplicate exclusions for some reason which used to cause issues with migrator. That's fixed now.

And the actual policy changes:

- ROX-18978: Updates two policies that check for `iptables` process to also check `nft` (which is for nftables)
- ROX-20225: Exclude deployment `aide-worker-fileintegrity` in namespace `openshift-file-integrity` from the policy "Process with UID 0"

If you are only interested in reviewing the policy, I recommend looking at only the files in the pkg/default/policies/files directory since that's the actual policies. Everything else is scaffolding for migrations.

## Checklist
~[ ] Evaluated and added CHANGELOG entry if required~
~[ ] Determined and documented upgrade steps~
~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

Manual testing

1. Installed fresh version of ACS and validated all the changes
2. Start with 4.2.2 and then upgraded to this build. Validated all the changes
3. Start with 4.2.12 Upgrade to this build. Rollback to 4.2.2. Roll forward to this version.

For #3 the most important thing is that central didn't error out or crash. On revert, all of the changes (new criteria, and exclusion) all remained. But that should be ok. On roll forward, the policy criteria remain. The exclusion unfortunately generates a duplicate, but I think that's a side effect we can take (especially since some existing policies already have dupes for exclusions). 

![Screenshot 2023-10-26 at 5 01 25 PM](https://github.com/stackrox/stackrox/assets/1407686/efa977d9-32aa-4844-8d74-3310f96548a3)



In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.
